src/java.prefs/unix/classes/java/util/prefs/FileSystemPreferences.java

```diff
@@ -701,16 +702,40 @@ protected AbstractPreferences childSpi(String name) {
         return new FileSystemPreferences(this, name);
     }
 
+    private static String getErrorString(int errCode) {
+        String os = System.getProperty("os.name");
+        if (errCode == 35 && (os.contains("Mac") || os.contains("Darwin"))) return "EAGAIN";
+        switch (errCode) {
+            case 13: return "EACCES";
+            case 11: return "EAGAIN"; // macOS has for this 35
+            case 9:  return "EBADF";
+            case 16: return "EBUSY";
+            case 17: return "EEXIST";
+            case 14: return "EFAULT";
+            case 4:  return "EINTR";
+            case 22: return "EINVAL";
+            case 24: return "EMFILE";
+            case 19: return "ENODEV";
+            case 2:  return "ENOENT";
+            case 1:  return "EPERM";
+        }
+        return "";
```

Comment:
I assume the use of System.getProperty is problematic when running with a SM.

In any case, I don't think this is the right way to do it, instead you want a native method to call strerror to translate the error.